src/modules/document-caching/index.ts

```diff
@@ -30,6 +30,7 @@ export type DocumentCachingOptions = {
   timeoutSeconds?: Number;
   maxDocumentsInCache?: Number;
   maxAgeSecondsforDocumentsInCache?: Number;
+  allowedNonAMPPages?: Array<RegExp>;
```

Comment:
Investigate extending the Plugin interfaces to create the DocumentCachingOptions interface.

Reply:
The reason why we cannot do this is because our keys do not match with the Plugin's config keys.
e.g. MaxExtries makes sense for workbox but for us the same key is known as `MaxDocumentsInCache` so the `extends` wouldn't work here

Reply:
done